cmd/certsuite/upload/results_spreadsheet/results_spreadsheet.go

```diff
@@ -0,0 +1,367 @@
+package resultsspreadsheet
+
+import (
+	"context"
+	"encoding/csv"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/drive/v3"
+	"google.golang.org/api/option"
+	"google.golang.org/api/sheets/v4"
+)
+
+var (
+	resultsFilePath string
+	rootFolderURL   string
+	ocpVersion      string
+)
+
+var workloadsResultsFolder *drive.File
+var stringToPointer = func(s string) *string { return &s }
+
+const (
+	categoryConclusionsCol        = "Category"
+	workloadVersionConclusionsCol = "Workload Version"
+	ocpVersionConclusionsCol      = "OCP Version"
+	workloadNameConclusionsCol    = "Workload Name"
+	resultsConclusionsCol         = "Results"
+	cellContantLimit              = 50000
+)
+
+var conclusionSheetHeaders = []string{categoryConclusionsCol, workloadVersionConclusionsCol, ocpVersionConclusionsCol, workloadNameConclusionsCol, resultsConclusionsCol}
+
+var (
+	uploadResultSpreadSheetCmd = &cobra.Command{
+		Use:   "results-spreadsheet",
+		Short: "Generates a google spread sheets with test suite results.",
+		Run: func(cmd *cobra.Command, args []string) {
+			generateResultsSpreadSheet()
+		},
+	}
+)
+
+func NewCommand() *cobra.Command {
+	uploadResultSpreadSheetCmd.Flags().StringVarP(&resultsFilePath, "results-file", "f", "", "Required: path to results file")
+	uploadResultSpreadSheetCmd.Flags().StringVarP(&rootFolderURL, "dest-url", "d", "", "Required: Destination drive folder's URL")
+	uploadResultSpreadSheetCmd.Flags().StringVarP(&ocpVersion, "version", "v", "", "Optional: OCP Version")
+
+	err := uploadResultSpreadSheetCmd.MarkFlagRequired("results-file")
+	if err != nil {
+		log.Fatalf("Failed to mark results file path as required parameter: %v", err)
+		return nil
+	}
+
+	err = uploadResultSpreadSheetCmd.MarkFlagRequired("dest-url")
+	if err != nil {
+		log.Fatalf("Failed to mark dest url path as required parameter: %v", err)
+		return nil
+	}
+
+	return uploadResultSpreadSheetCmd
+}
+
+func readCSV(fp string) ([][]string, error) {
+	file, err := os.Open(fp)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	reader := csv.NewReader(file)
+	records, err := reader.ReadAll()
+	if err != nil {
+		return nil, err
+	}
+	return records, nil
+}
+
+func createSheetsAndDriveServices() (sheetService *sheets.Service, driveService *drive.Service, err error) {
+	ctx := context.Background()
+	b, err := os.ReadFile("cmd/certsuite/upload/results_spreadsheet/credentials.json")
```

Comment:
Why not expect credentials.json and token.json in the same folder as the executable? Maybe optionally pass a credentials folder path?

Reply:
The token is created if does not exists, but the credentials can be passed if we want. I though it might be better to store the credentials and the token in the same folder as the sub-command to give it more context, but they can be moved to same folder as the executable.
What do you thinks?

Reply:
I would prefer to use same folder as the executable or a configured folder optionally. The folder of the subcommand does not have a meaning when running the executable, it is just relevant at build time to organize the code. So if running the exe outside of our repo it would require a very specific directory to be created.